components/admin/lessons/AdminLessonSideNav.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect } from 'react'
+import { ApolloError } from '@apollo/client'
+import { AlertFillIcon, ChevronRightIcon } from '@primer/octicons-react'
+import { isNumber } from 'lodash'
+import { Spinner } from 'react-bootstrap'
+import { get } from 'lodash'
+import styles from '../../../scss/adminLessonSideNav.module.scss'
+
+type Item = {
+  id: number
+  name: string
+  content: string
+  lesson: { id?: number }
+}
+
+type Items = Item[]
+
+type Props = {
+  items: Items | undefined
+  title: string
+  lessonId: number
+  onAddItem: (m: Item | undefined) => void
+  loading?: boolean
+  error?: ApolloError
+  setActive?: React.Dispatch<React.SetStateAction<number>>
+  active?: number
+}
+
+const AdminLessonSideNav = ({
+  items,
+  onAddItem,
+  loading,
+  error,
+  setActive,
+  active,
+  title,
+  lessonId
+}: Props) => {
+  const setAsActive = (id: number) => setActive && setActive(id)
+
+  const itemsList =
+    items &&
+    items.reduce((acc: JSX.Element[], item) => {
```

Comment:
I think `map` is more appropriate. if there are 5 items, there would be 5 elements

```js
const itemsList = (items || []).map(...)
```

Reply:
My initial approach was with `map`, but its array will end up having some `undefined` elements because it's filtering them. Using `filter` with `map` is O(a + b) or 2 loops with each one being O(n). The best solution I found is `reduce`.

Reply:
the filtering should be done in the parent and only pass in the relevant lessons into this component.